packages/espressocash_app/lib/features/incoming_split_key_payments/src/widgets/pending_iskp_listener.dart

```diff
@@ -30,16 +30,19 @@ class _PendingISKPListenerState extends State<PendingISKPListener> {
     if (existing != null) {
       if (!mounted) return;
 
-      _openFirstPartReadyScreen();
+      await _openFirstPartReadyScreen();
     }
   }
 
-  void _openFirstPartReadyScreen() {
-    context.router.push(
-      FirstPartReadyRoute(
-        onCancel: () => sl<PendingISKPRepository>().clear(),
-      ),
-    );
+  Future<void> _openFirstPartReadyScreen() async {
+    if (!mounted) return;
```

Comment:
Looks weird? Why do you need to check for mounted as the first call in the method?